File: src/main/java/akostenko/aicars/track/CircularTrack.java
package akostenko.aicars.track;

import java.util.List;

public class CircularTrack extends Track {

    static final String NAME = "Circular";
    private final int width = 12;

    private List<TrackSection> sections = TrackBuilder.start(0, 0, 0, width)
            .right(300, 180)
            .right(300, 180)
            .done();

    @Override
    public String getTitle() {
        return NAME;
    }

    @Override
    public double getWidth() {
        return width;
    }

    @Override
    public List<TrackSection> sections() {
        return sections;
    }
}

File: src/main/java/akostenko/aicars/track/CircularTrack.kt
package akostenko.aicars.track

class CircularTrack : Track() {
    override val width = 16.0

    override val sections = TrackBuilder.start(0.0, 0.0, 0.0, width)
            .right(300.0, 180.0)
            .right(300.0, 180.0)
            .done()

    override val title = NAME

    companion object {
        internal val NAME = "Circular"
    }
}
